### DISCUSSION TITLE
Issue: Failed connection to server on mobile with "connect_error due to xhr poll error"

### DISCUSSION THREAD

**Discussion by EhsanKey** [socketio/socket.io#4764] [Q&A]
Hello everyone!

I have developed a project using `socket-io-client` (Next.js) on the client side and `socket-io` (Node.js, Express) on the server side. When I run the project on my computer's browser, everything works fine. However, when I try to run the project on a mobile browser, I encounter a connection issue with the following error message: "connect_error due to xhr poll error".

The problem arises during the connection to the server, and the specific error message is as follows:

```
polling.js:307 GET http://localhost:4000/socket.io/?EIO=4&transport=polling&t=OavGB5H net::ERR_CONNECTION_REFUSED
```

Here is the client-side code:

```javascript
socket.start("http://localhost:4000/", {
  autoConnect: false,
  auth: {
    token: getCookie("token"),
  },
});

socket.connect((res) => {
  console.log(res, "socket connected");
});

socket.on("connect_error", (err) => {
  console.log(`connect_error due to ${err.message}`);
});

socket.on("connect", () => {
  console.log("socket connected");
});

socket.on("disconnect", () => {
  console.log("socket disconnected");
});
```

And here is the server-side code:

```javascript
const express = require("express");
const http = require("http");
const socketIO = require("socket.io");
const app = express();
const server = http.createServer(app);

const io = socketIO(server, {
  cors: { origin: "*" },
});

// Other code and configurations

io.on("connection", (socket) => {
  console.log("connected");
});

const port = 4000;
server.listen(port, () => {
  console.log(`Server is listening on port ${port}`);
});
```

If you have any experience or a solution regarding this issue, please share it in this discussion.

Thank you!

**Comment by darrachequesne** [comment]
Hi! I think that's because you can't use `localhost` to reach your server on your mobile.

You'll need to be on the same network as your server (wifi) and use the IP address of your server, something like `192.168.0.10`.

**Reply by EhsanKey** [reply]
I have disabled the firewall, and I can view my project on a mobile browser. However, the problem is that it cannot connect to the socket!
